controllers/reconcilers/kserve_istio_podmonitor_reconciler.go

```diff
@@ -95,6 +99,50 @@ func (r *KserveIstioPodMonitorReconciler) createDesiredResource(isvc *kservev1be
 				{
 					Path:     "/stats/prometheus",
 					Interval: "30s",
+					RelabelConfigs: []*v1.RelabelConfig{
```

Comment:
Thinking more about this `PodMonitor` now, I am wondering if we need to add a labelselector to it. 
Consider scenario: Namespace has single model serving deployed along with other workloads (notebooks for example)
Here, I think the Podmonitor would end up creating an aggregate endpoint target even for the notebook pods which would end up being `Down`. I don't know if this even has the possibility to break anything for the other workloads (seems unlikely). But something to consider. 

Adding the labelselector however, does not have a downside even if no other workloads are present. We are just ensuring we don't create unnecessary Targets in prometheus (which would be broken anyway). 

cc @spolti @israel-hdez wdyt?

Reply:
I guess you are talking about improving line 90, rather than 102?
If so, your comment makes sense to me.

Reply:
Yeah the location of the comment wasn't specific to what I was talking about here 😅 
Just want the PodMonitor to have a labelselector on it for `component: predictor` probably

Reply:
Yes, that makes sense. Otherwise, this can be selecting too many pods.

Reply:
In this case, don't we also want to have explainers and transformers?
using only predictors might lead to problems if we also need to fetch metrics from the explainers in the future.

For this, I am using the label selector as In, and the values: predictor, explainer or transformer:

```yaml
  selector:
    matchExpressions:
      - key: istio-prometheus-ignore
        operator: In
        values:
        - predictor
        - explainer
        - transformer

```

Reply:
with this setup it is also working and limited to objects that have the `component: predictor| explainer|transformer` label.